annubes/task.py

```diff
@@ -352,48 +353,54 @@ def _check_time_vars(self) -> None:
 
     def _build_trials_seq(self) -> NDArray[np.str_]:
         """Generate a sequence of modalities."""
-        # Extract keys and probabilities from the dictionary
-        scenarios = list(self._session.keys())
-        probabilities = np.array(list(self._session.values()))
-        # Generate random numbers of samples based on the probabilities
-        prob_samples = self._rng.multinomial(self._ntrials, probabilities)
-        # Create a dictionary to store the results
-        session_in_samples = {
-            scenario: self._rng.multinomial(prob_samples[i], [1 - self.catch_prob, self.catch_prob])
-            for i, scenario in enumerate(scenarios)
-        }
-        # Generate the sequence of modalities
-        modality_seq = []
-        for m in scenarios:
-            temp_seq = session_in_samples[m][0] * [m] + session_in_samples[m][1] * ["catch"]
-            self._rng.shuffle(temp_seq)
-            modality_seq += list(temp_seq)
-        if self.shuffle_trials:
-            self._rng.shuffle(modality_seq)
-            if self.max_sequential:
-                # Shuffle the list using Fisher-Yates algorithm with consecutive constraint
-                i = len(modality_seq) - 1
-                while i > 0:
-                    # Picking j can't be fixed, otherwise the algorithm is not random
-                    # We may want to change this in the future
-                    j = self._rng.integers(0, i)
-                    modality_seq[i], modality_seq[j] = modality_seq[j], modality_seq[i]
-                    i -= 1
-                    # Check and fix the consecutive constraint
-                    count = 1
-                    while i > 0 and modality_seq[i] == modality_seq[i - 1] and count >= self.max_sequential:
-                        i -= 1
-                joined_seq = "".join(modality_seq)
-                for mod in self._session:
-                    violations = joined_seq.count(mod * (self.max_sequential + 1))
-                    if violations > 0:
-                        warnings.warn(
-                            f"`max_sequential` limit of {self.max_sequential} was violated {violations} times"
-                            f" for {mod}.\n"
-                            "Please check the current trials sequence, and if it is not acceptable redefine your"
-                            f"task or try to change the random seed, now set to {self._random_seed}.\n",
-                            stacklevel=2,
-                        )
+        # Extract options and probs from the dictionary
+        options = list(self._session.keys())
+        probs = list(self._session.values())
+        max_seq = self.max_sequential
+
+        if not (self.shuffle_trials and max_seq):
```

Comment:
Indeed clear the list would be the way, but I do agree that feels weird. I think it feels already better now with the `self._constrained_shuffle` attribute, but I'd swap the if else statement: instead of starting with the `if not` condition, I'd place the shuffle + max_seq first, and then the else.

Reply:
For readability, I think it makes more sense to leave it as it is, despite the negative conditional.

Now the order (in the code base) is:
1. unshuffled
2. simple shuffle
3. constrained shuffle.

If I would turn the condition around the order would be: 
1. constrained shuffle
2. unshuffled
3. simple shuffle.

Reply:
Ok let's keep it this way